joshua/joshua_model.py

```diff
@@ -978,3 +991,14 @@ def get_agent_failures(tr, time_start=None, time_end=None):
         failures.append((info, msg))
 
     return failures
+
+@transactional
+def cancel_agent_cleanup(tr, ensemble_id):
+    """
+    Clean-up method for when an agent takes a job but gets cancelled
+    """
+
+    # TODO(qhoang) let's try this but there must be a better way
+    # When an agent is cancelled, it has already incremented the __started__ counter
+    # but will never get to increment the __ended__ counter
+    _decrement(tr, ensemble_id, "started")
```

Comment:
This needs to be idempotent, see https://apple.github.io/foundationdb/developer-guide.html#transactions-with-unknown-results